src/Microsoft.ML.ImageAnalytics/ImageLoader.cs

```diff
@@ -201,11 +242,107 @@ protected override Delegate MakeGetter(DataViewRow input, int iinfo, Func<int, b
                                 throw Host.Except($"Failed to load image {src.ToString()}.");
                         }
                     };
+
+                return del;
+            }
+
+            private Delegate MakeGetterVectorDataViewByteType(DataViewRow input, int iinfo, Func<int, bool> activeOutput, out Action disposer)
+            {
+                Contracts.AssertValue(input);
+                Contracts.Assert(0 <= iinfo && iinfo < _parent.ColumnPairs.Length);
+
+                disposer = null;
+                var getSrc = input.GetGetter<ReadOnlyMemory<char>>(input.Schema[ColMapNewToOld[iinfo]]);
+                ReadOnlyMemory<char> src = default;
+                ValueGetter<VBuffer<byte>> del =
+                    (ref VBuffer<byte> dst) =>
+                    {
+                        byte[] buffer = null;
+                        bool addBufferBack = false;
+                        if (!_bufferPool.TryTake(out buffer))
+                        {
+                            buffer = new byte[4096];
+                            addBufferBack = true;
+                        }
+
+                        getSrc(ref src);
+
+                        if (src.Length > 0)
+                        {
+                            string path = src.ToString();
+                            if (!string.IsNullOrWhiteSpace(_parent.ImageFolder))
+                                path = Path.Combine(_parent.ImageFolder, path);
+                            if (!TryLoadDataIntoBuffer(path, ref dst, buffer))
+                                throw Host.Except($"Failed to load image {src.ToString()}.");
+                        }
+
+                        if (addBufferBack)
+                        {
+                            Contract.Assert(buffer != null);
+                            _bufferPool.Add(buffer);
+                        }
```

Comment:
Replace this with just _bufferPool.Add(buffer); so we always put the buffer back into the pool and don't throw it away. Leave the assert in there